File: tgbot/src/main/java/com/example/entities/Event.java
package com.example.entities;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;

import com.example.entities.User.User;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "events")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String description;

    private Date startTime;

    private Date endTime;

    private int numberSeats;

    private int maxSeats;

    private String location;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organizer_id")
    private User organizer;

    @ManyToMany(mappedBy = "events")
    private Set<User> participants = new HashSet<>();

    @ManyToMany(mappedBy = "favoriteEvents")
    private Set<User> favoritedByUsers = new HashSet<>();

}

File: frontend/my-vite-app/src/types/event.ts
export interface Event {
  id: number;
  name: string;
  description: string;
  startTime: string;
  endTime: string;
  location?: string;
  numberSeats: number;
  maxSeats: number;
  organizerId: number;
}

export interface EventCardProps {
  title: string;
  description: string;
  date: string;
  fullDate: string;
  location?: string;
  spotsLeft: number;
  imageUrl?: string;
  category?: string;
  timeline?: Array<{
    time: string;
    title: string;
    description: string;
  }>;
}